### PR TITLE
Switch to Tox

### PR DESCRIPTION
**Pull request by stephendonner**
Fixes https://github.com/mozilla/snippets-tests/issues/56

Ad-hoc job results: http://pastebin.com/GzfYBb2K

@mozilla/web-qa-apprentices / @mozilla/web-qa-sorcerers r?

### CODE REVIEW

**Comment by stephendonner** [comment]
This will need just a couple simple tweaks + line removals in Jenkins, after it's merged, so either I or someone else should be around to fix, just FYI.

**Comment by m8ttyB** [comment]
r+ lg[reat]tm and tests appear to run as expected.

**Comment by stephendonner** [comment]
Thx @m8ttyB <3

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 sudo: false
 language: python
 python: 2.7
-install: "pip install flake8"
-script: "flake8 ."
+env:
+  - TOXENV=tests
```
Comment (davehunt) [inline .travis.yml]: This means we've switched from running flake8 to running the tests. Was this intentional? We should at least run flake8 in Travis and carefully consider if we want to run the tests (we do that in Jenkins, too).

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+[testenv]
+passenv = PYTEST_ADDOPTS
+deps =
+    beautifulsoup4==4.5.1
+    pytest~=3.0.0
+    pytest-base-url
+    pytest-xdist==1.15.0
+    requests==2.11.1
+commands = py.test {posargs}
```
Comment (davehunt) [inline tox.ini]: I'd suggest using `pytest` now that we're using pytest 3.

File: README.md [hunk 3]
```diff
@@ -12,50 +12,8 @@ Automated tests for the Snippets web app
...
-to install the required Python libraries.
-
-### Running Tests
-Once the above prerequisites have been met, you can run the tests using the
-following command:
-
-    py.test tests/test_snippets.py
-
-For other possible options, type `py.test --help`.
+* `Install Tox <https://tox.readthedocs.io/en/latest/install.html>`_
```
Comment (davehunt) [inline README.md]: This isn't a valid markdown link. It's being rendered as a code snippet.

File: tox.ini (hunk 2 above)
Comment (stephendonner) [inline tox.ini]: Sorry, yes - I'll fix this.  I should've remembered.

File: README.md (hunk 3 above)
Comment (stephendonner) [inline README.md]: Will fix.

File: .travis.yml (hunk 1 above)
Comment (stephendonner) [inline .travis.yml]: Slip-up; will fix in an upcoming patch.  Wasn't intending to run the tests in Travis CI, but definitely meant to retain running flake8 there.

**Pull request merged**